embedder/embedder.go

```diff
@@ -122,29 +123,58 @@ func (flu *FlutterEngine) Run(userData unsafe.Pointer, vmArgs []string) Result {
 	defer C.free(unsafe.Pointer(icuDataPath))
 
 	args := C.FlutterProjectArgs{
-		assets_path:       assetsPath,
-		icu_data_path:     icuDataPath,
-		command_line_argv: (**C.char)(cVMArgs),
-		command_line_argc: C.int(len(vmArgs)),
+		assets_path:                assetsPath,
+		icu_data_path:              icuDataPath,
+		command_line_argv:          (**C.char)(cVMArgs),
+		command_line_argc:          C.int(len(vmArgs)),
+		shutdown_dart_vm_when_done: true,
+	}
+
+	if C.FlutterEngineRunsAOTCompiledDartCode() {
+		// TODO //
+		// https://github.com/flutter/engine/pull/18146
+		/////////
+
+		// elfSnapshotPath := C.CString(flu.ElfSnapshotPath)
+		// defer C.free(unsafe.Pointer(elfSnapshotPath))
+
+		// dataIn := C.FlutterEngineAOTDataSource{
+		// _type:    C.kFlutterEngineAOTDataSourceTypeElfPath,
+		// elf_path: elfSnapshotPath,
+		// }
+		// res := C.FlutterEngineCreateAOTData(&dataIn, &flu.aotDataSource)
+		// if res != ResultSuccess {
+		// return res.GoError("C.FlutterEngineCreateAOTData()")
+		// }
+		// args.aot_data = flu.aotDataSource
 	}
 
 	args.struct_size = C.size_t(unsafe.Sizeof(args))
 
-	res := C.runFlutter(userData, &flu.Engine, &args)
+	res := (Result)(C.runFlutter(userData, &flu.Engine, &args))
 	if flu.Engine == nil {
-		return ResultInvalidArguments
+		return ResultInvalidArguments.GoError("engine.Run()")
 	}
 
-	return (Result)(res)
+	return res.GoError("engine.Run()")
 }
 
 // Shutdown stops the Flutter engine.
-func (flu *FlutterEngine) Shutdown() Result {
+func (flu *FlutterEngine) Shutdown() error {
 	flu.sync.Lock()
 	defer flu.sync.Unlock()
 	flu.closed = true
+	defer func() {
+		if C.FlutterEngineRunsAOTCompiledDartCode() {
+			// res := C.FlutterEngineCollectAOTData(flu.aotDataSource)
```

Comment:
should this error be returned if it occurs instead of just logging it out (in the commented out code)